src/main/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJob.java

```diff
@@ -46,8 +49,15 @@ public void run() {
                 .map(this::tryParse)
                 .filter(Objects::nonNull)
                 .forEach(parsedReport -> {
-                    parsedReport.statuses.forEach(sendLetterClient::updatePrintedAt);
-                    ftpClient.deleteReport(parsedReport.path);
+                    List<Boolean> successStatuses = parsedReport
+                        .statuses
+                        .stream()
+                        .map(this::trySendUpdate)
+                        .collect(toList());
```

Comment:
would it be better to just continue with the stream here avoiding collection and essentially evaluation of the list?

Reply:
I can't, `allMatch` will stop going over the stream on first `false`

Reply:
😞. how about using `.reduce` function instead of `.map`?